client/ayon_applications/plugins/launcher_actions/debug_shell.py

```diff
@@ -0,0 +1,175 @@
+import os
+from typing import Optional
+
+from qtpy import QtWidgets, QtGui, QtCore
+
+from ayon_applications import (
+    Application,
+    ApplicationManager,
+    APPLICATIONS_ADDON_ROOT
+)
+from ayon_applications.utils import get_app_environments_for_context
+from ayon_core.pipeline import LauncherAction
+from ayon_core.style import load_stylesheet
+from ayon_core.tools.utils.lib import get_qt_icon
+
+
+def get_application_qt_icon(application: Application) -> Optional[QtGui.QIcon]:
+    """Return QtGui.QIcon for an Application
+
+    Note: This forces the icons to be search in `ayon_applications/icons`
+        folder on client side and mimics what Ayon applications addon's
+        `get_app_icon_path` method does.
+    """
+    icon = application.icon
+    if not icon:
+        return QtGui.QIcon()
+    icon_filename = os.path.basename(icon)
+    icon_filepath = os.path.join(
+        APPLICATIONS_ADDON_ROOT, "icons", icon_filename)
+    if os.path.exists(icon_filepath):
+        return get_qt_icon({"type": "path", "path": icon_filepath})
+    return QtGui.QIcon()
+
+
+class DebugShell(LauncherAction):
+    """Run any host environment in command line."""
+    name = "debugshell"
+    label = "Shell"
+    icon = "terminal"
+    color = "#e8770e"
+    order = 10
+
+    def is_compatible(self, selection) -> bool:
+        return selection.is_task_selected
+
+    def process(self, selection, **kwargs):
+        # Get cursor position directly so the menu shows closer to where user
+        # clicked because the get applications logic might take a brief moment
+        pos = QtGui.QCursor.pos()
+        application_manager = ApplicationManager()
+
+        # Choose shell
+        shell_applications = self.get_shell_applications(application_manager)
+        if len(shell_applications) == 0:
+            raise RuntimeError(
```

Comment:
`RuntimeError` is kinda cruel.
```suggestion
            raise ValueError(
```